File: src/main/java/icu/takeneko/omms/client/session/data/Request.java
package icu.takeneko.omms.client.session.data;

import lombok.Setter;
import lombok.ToString;

import java.util.HashMap;

@ToString
public class Request {
    public Request(String req) {
        this.request = req;
    }

    public Request() {
    }

    private String request;
    private HashMap<String, String> content = new HashMap<>();
    @Setter
    private String requestId;

    public String getContent(String key) {
        return content.get(key);
    }

    public Request setRequest(String request) {
        this.request = request;
        return this;
    }

    public Request withContentKeyPair(String key, String pair) {
        content.put(key, pair);
        return this;
    }
}

File: src/main/java/icu/takeneko/omms/client/session/request/Request.kt
package icu.takeneko.omms.client.session.request

open class Request() {
    constructor(request: String) : this() {
        this.request = request
    }
    lateinit var request: String
        protected set

    fun setRequest(request: String): Request {
        this.request = request
        return this
    }

    var content: MutableMap<String, String> = mutableMapOf()

    fun withContentKeyPair(key: String, value: String): Request {
        content[key] = value
        return this
    }

    override fun toString(): String = "Request { request=$request, content=$content }"
}
